testing/test_cacheprovider.py

```diff
@@ -300,22 +278,8 @@ def test_3():
         result.stdout.fnmatch_lines(["*1 failed*2 passed*"])
 
     def test_failedfirst_order(self, testdir):
-        testdir.tmpdir.join("test_a.py").write(
-            textwrap.dedent(
-                """\
-                def test_always_passes():
-                    assert 1
-                """
-            )
-        )
-        testdir.tmpdir.join("test_b.py").write(
-            textwrap.dedent(
-                """\
-                def test_always_fails():
-                    assert 0
-                """
-            )
-        )
+        testdir.makepyfile(**{"test_a.py": "def test_always_passes(): pass"})
```

Comment:
Is the `.py` in `test_a.py` still needed with `makepyfile` (rather than `makefile`)?

Reply:
No, and can also use kwargs directly.  Amended.